apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -36,48 +38,79 @@ public class ApolloIRGenTask extends NodeTask {
     variant = variantName
     config = extensionsConfig
     group = ApolloPlugin.TASK_GROUP
-    description = "Generate IR files using apollo-codgen for ${variant.capitalize()} GraphQL queries"
+    description = "Generate an IR file using apollo-codgen for ${variant.capitalize()} GraphQL queries"
     dependsOn(ApolloCodeGenInstallTask.NAME)
-    outputDir = new File("${getSchemaPath()}/generatedIR")
+
+    possibleGraphQLPaths = buildPossibleGraphQLPaths()
+    outputDir = project.file("src/$variant/graphql/generatedIR")
```

Comment:
We can no longer depend on user provided options to decide the output directory. Because, we have the `src/main/graphql` check in the compiler, so compilation would fail if we generate the IR in a violating directory.

The IR will always be generated under `src/VariantName/graphql/generatedIR`. 

**The variant name is the build variant source set which is the product flavor plus build type**